### PR TITLE
#167619168 Create Sessions Table

### PR DESCRIPTION
**Pull request by ayodejiAA**
#### What does this PR do?
- Create migration for session table
- Add demo seed file

#### Description of Task proposed in this pull request?
-  Add session migration file
- Add Session model
- Add Session seed file

#### How should this be manually tested (Quality Assurance)?
- Check out the branch
-Add the database connection string in .env file
- Run ```npm run db:seed```

#### What are the relevant pivotal tracker stories?
- Story ID: [#167619168](https://www.pivotaltracker.com/story/show/167619168)

#### Any background context you want to add (Operations Impact)?
- Requires database connection string for it to work

#### What I have learned working on this feature:
- Use sequelize cli to generate migrations and seeds

#### Screenshots: [If you made some visual changes to the application please upload screenshots here, or remove this section]
- n/a

### CODE REVIEW

File: server/database/migrations/20190731135151-create-session.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    ipAddress: {
+      allowNull: false,
+      type: Sequelize.STRING
+    },
+    token: {
+      allowNull: false,
+      type: Sequelize.STRING
+    },
+    userAgent: {
+      allowNull: false,
```
Comment (dinobi) [inline server/database/migrations/20190731135151-create-session.js]: allow null here

File: server/database/migrations/20190731135151-create-session.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      type: Sequelize.STRING,
+      defaultValue: 'browser'
+    },
+    expiresAt: {
+      allowNull: false,
+      type: Sequelize.DATE
+    },
+    ipAddress: {
+      allowNull: false,
```
Comment (dinobi) [inline server/database/migrations/20190731135151-create-session.js]: allow null here

File: server/database/models/session.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+module.exports = (sequelize, DataTypes) => {
+  const Session = sequelize.define(
+    'Sessions',
+    {
+      userId: DataTypes.INTEGER,
+      active: DataTypes.BOOLEAN,
+      devicePlatform: DataTypes.STRING,
+      expiresAt: DataTypes.DATE,
```
Comment (dinobi) [inline server/database/models/session.js:8]: Don't sequelize have a date time type?

Comment (dinobi) [reply server/database/models/session.js:8]: this is datetime, confirmed.

**Review by dinobi — CHANGES_REQUESTED** [review]
Please add model tests

**Comment by ayodejiAA** [comment]
> Please add model tests

Okay. On it.

File: server/database/migrations/20190731135151-create-session.js (hunk 2 above)
Comment (ayodejiAA) [reply server/database/migrations/20190731135151-create-session.js]: Done

File: server/database/migrations/20190731135151-create-session.js (hunk 1 above)
Comment (ayodejiAA) [reply server/database/migrations/20190731135151-create-session.js]: Done

**Comment by ayodejiAA** [comment]
> Please add model tests
@dinobi  Implemented the model test

**Review by henryade — DISMISSED** [review]

**Review by Adekoreday — DISMISSED** [review]

File: server/database/migrations/20190731135151-create-session.js (hunk 2 above)
Comment (chialuka) [reply server/database/migrations/20190731135151-create-session.js]: I think allow null here means that the field should be nullable, or am I wrong?

File: server/database/migrations/20190731135151-create-session.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+module.exports = {
```
Comment (chialuka) [inline server/database/migrations/20190731135151-create-session.js]: Please consider changing this to ES6 exports

Comment (ayodejiAA) [reply server/database/migrations/20190731135151-create-session.js]: Done

**Review by dinobi — APPROVED** [review]

**Pull request merged**